### PR TITLE
Add ARM build logic

### PR DESCRIPTION
**Pull request by DarcyRaynerDD**
Adds Graviton 2 publishing support to Lambda Extension

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ jobs:
       - name: Install Serverless Framework
         run: sudo yarn global add serverless --prefix /usr/local
 
+        - name: Install Crossbuild Deps
```
Comment (nhinsch) [inline .github/workflows/integration_tests.yml:35]: YML indentation looks off here

File: scripts/build_binary_and_layer_dockerized.sh [hunk 2]
```diff
@@ -38,9 +38,19 @@ cd $AGENT_PATH/..
 tar --exclude=.git -czf $BASE_PATH/scripts/.src/datadog-agent.tgz datadog-agent
 cd $BASE_PATH
 
-DOCKER_BUILDKIT=1 docker build -t datadog/build-lambda-extension:$VERSION \
-    -f ./scripts/Dockerfile.build \
-    --build-arg EXTENSION_VERSION="${VERSION}" .
+function docker_build_zip {
+    arch=$1
+
+    DOCKER_BUILDKIT=1 docker buildx build --platform linux/${arch} \
```
Comment (maxday) [inline scripts/build_binary_and_layer_dockerized.sh:44]: DOCKER_BUILDKIT=1 is not required anymore if buildx is used

File: scripts/build_docker_image.sh [hunk 3]
```diff
@@ -27,10 +27,10 @@ unzip .layers/datadog_extension.zip
...
 echo "Building the Docker image"
-docker build extensions \
-  -f $DOCKERFILE_LOCATION \
-  -t $DOCKER_REPOSITORY_NAME:$VERSION \
-  --no-cache
 
-# Also tag the image with :latest
-docker tag $DOCKER_REPOSITORY_NAME:$VERSION $DOCKER_REPOSITORY_NAME:latest
\ No newline at end of file
+DOCKER_BUILDKIT=1 docker buildx build --platform linux/amd64,linux/arm64 \
```
Comment (maxday) [inline scripts/build_docker_image.sh:31]: same here

File: scripts/build_docker_image.sh [hunk 4]
```diff
@@ -27,10 +27,10 @@ unzip .layers/datadog_extension.zip
 
 # Build the image, tagged with the version
 echo "Building the Docker image"
```
Comment (nhinsch) [inline scripts/build_docker_image.sh:29]: It looks like this script now builds *and* pushes and the Docker image(s), I would rename it to reflect that, and also maybe update this `echo` message to make that more clear

File: scripts/publish_prod.sh [hunk 5]
```diff
@@ -77,8 +76,7 @@ saml2aws login -a govcloud-us1-fed-human-engineering
 AWS_PROFILE=govcloud-us1-fed-human-engineering ./scripts/publish_layers.sh
 
 echo "Pushing Docker image to Dockerhub"
```
Comment (nhinsch) [inline scripts/publish_prod.sh:78]: I would update or remove this `echo` statement

**Review by nhinsch — APPROVED** [review]
Looks good! Just some nits

**Comment by maxday** [comment]
LGTM as well (I was able to build both locally)

**Review by maxday — APPROVED** [review]

**Pull request merged**